src/retrace/retrace.py

```diff
@@ -1310,6 +1310,55 @@ def package_name(self, debug=False):
     def needs_arch(self):
         return self._arch is None
 
+class KernelVMcore:
+    def __init__(self, vmcore_path):
+        self._vmcore_path = vmcore_path
+        self._is_flattened = None
+
+    def is_flattened_format(self):
+        """Returns True if vmcore is in makedumpfile flattened format"""
+        if self._is_flattened is not None:
+            return self._is_flattened
+        try:
+            with open(self._vmcore_path) as fd:
+                fd.seek(0)
+                # Read 16 bytes (SIG_LEN_MDF from crash-utility makedumpfile.h)
+                b = os.read(fd.fileno(), 16)
+                fd.close()
```

Comment:
The point of using `with` is that you don't need to close the file manually now. So you can remove this line as well as the analogue in `convert_flattened_format()`.

Reply:
fixed in 31eafd8